tests/python/frontend/onnx/test_forward.py

```diff
@@ -77,22 +77,27 @@ def get_tvm_output(graph_def, input_data, target, ctx, output_shape=None, output
         return tvm_output.asnumpy()
 
 
-def get_caffe2_output(model, x, dtype='float32'):
-    import caffe2.python.onnx.backend
-    prepared_backend = caffe2.python.onnx.backend.prepare(model)
-    W = {model.graph.input[0].name: x.astype(dtype)}
-    c2_out = prepared_backend.run(W)[0]
-    return c2_out
+def get_onnxruntime_output(model, x, dtype='float32'):
+    import onnxruntime.backend
+    rep = onnxruntime.backend.prepare(model, 'CPU')
+    x = x.astype(dtype)
+    ort_out = rep.run(x)[0]
+    return ort_out
 
 
 def verify_onnx_forward_impl(graph_file, data_shape, out_shape):
     dtype = 'float32'
     x = np.random.uniform(size=data_shape)
     model = onnx.load_model(graph_file)
-    c2_out = get_caffe2_output(model, x, dtype)
-    for target, ctx in ctx_list():
-        tvm_out = get_tvm_output(model, x, target, ctx, out_shape, dtype)
-        tvm.testing.assert_allclose(c2_out, tvm_out, rtol=1e-5, atol=1e-5)
+    try:
+        c2_out = get_onnxruntime_output(model, x, dtype)
+    except onnx.onnx_cpp2py_export.checker.ValidationError as e:
```

Comment:
Last one I promise, but I don't think we want to be doing try excepts here. If tests fail its important to throw an error so it can be fixed and won't sneak into the master branch. Did you need to add this because you were hitting ValidationErrors?

Reply:
Hi @jwfromm 

Thanks for the prompt reply.

I remove the most part of try excepts but `DepthToSpace` because there is a strange issue error here:

```
onnx.onnx_cpp2py_export.checker.ValidationError: Unrecognized attribute: mode for operator DepthToSpace

==> Context: Bad node spec: input: "x" output: "y" op_type: "DepthToSpace" attribute { name: "blocksize" i: 2 type: INT } attribute { name: "mode" s: "CRD" type: STRING }
```

I check the example from [onnx.operator.DepthToSpace](https://github.com/onnx/onnx/blob/master/docs/Operators.md#DepthToSpace) document and it has the `mode` attribute for `DepthToSpace`, therefore, I keep the try-catch for `DepthToSpace`, currently.

Reply:
Oh interesting. The checker must be using the OpSet-1 version of DepthToSpace, which doesn't have a mode argument. Maybe for our test we should just use the default mode argument (i.e. not specifying it).

Reply:
Hi @jwfromm 

Sorry for the late reply.

I remove the `try excepts` in the testing case and use the default mode argument for now.